### PR TITLE
[MDEP-752] Add ignoredPackagings options for analyze

### PR DESCRIPTION
**Pull request by slawekjaranowski**
By default pom and ear packaging are ignored from analyzing.

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MDEP) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MDEP-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MDEP-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by slachiewicz** [comment]
Our builds for Maven 3.2 fails https://ci-maven.apache.org/job/Maven/job/maven-box/job/maven-dependency-plugin/job/MDEP-752/
can You check it?

**Comment by slawekjaranowski** [comment]
When we have default value, like:

```
 @Parameter( defaultValue = "pom,ear" )
 private List<String> ignoredPackagings = new ArrayList<>();
```

In maven `3.2.5` this default value is used always and win with configuration for collection in pom.

fixed to:

```
@Parameter
private List<String> ignoredPackagings = Arrays.asList( "pom", "ear" );
```

**Comment by slawekjaranowski** [comment]
@slachiewicz please try again on jenkins

**Comment by pzygielo** [comment]
> When we have default value, like:
> 
> ```
>  @Parameter( defaultValue = "pom,ear" )
>  private List<String> ignoredPackagings = new ArrayList<>();
> ```
> 
> In maven `3.2.5` this default value is used always and win with configuration for collection in pom.
> 
> fixed to:
> 
> ```
> @Parameter
> private List<String> ignoredPackagings = Arrays.asList( "pom", "ear" );
> ```

But `defaultValue` finds its way into documentation, in standard way. With above (fixed to) approach I would definitely have missed that one sentence of description, when looking for defaults only. And maven 3.2.5 looks like very, very old version to support its bugs here.

File: src/main/java/org/apache/maven/plugins/dependency/analyze/AbstractAnalyzeMojo.java [hunk 1]
```diff
@@ -222,6 +222,18 @@
...
+     * List of project packaging that will be ignored.
+     *
+     * <p>
+     * by default <b>pom</b> and <b>ear</b> are ignored.
+     * </p>
+     *
+     * @since 3.2.1
+     */
+    @Parameter
+    private List<String> ignoredPackagings = Arrays.asList( "pom", "ear" );
```
Comment (pzygielo) [inline src/main/java/org/apache/maven/plugins/dependency/analyze/AbstractAnalyzeMojo.java:235]: I'd also like to see `rar` here.

Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/plugins/dependency/analyze/AbstractAnalyzeMojo.java:235]: - dock will be fixed.
- Maven 3.2.5 is used on ASF jenkins ... so we need workaround ...
- I prefer add new package with new issue and new PR - I don't want to change scope of this

Comment (pzygielo) [reply src/main/java/org/apache/maven/plugins/dependency/analyze/AbstractAnalyzeMojo.java:235]: >     * Maven 3.2.5 is used on ASF jenkins ... so we need workaround ...

:disappointed: 

>     * I prefer add new package with new issue and new PR - I don't want to change scope of this
:+1:

Comment (slawekjaranowski) [reply src/main/java/org/apache/maven/plugins/dependency/analyze/AbstractAnalyzeMojo.java:235]: Now look like standard defaults

image

**Comment by slawekjaranowski** [comment]
@slachiewicz please refresh branch for jenkins

**Review by slachiewicz — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #1266

**Pull request closed without merging**